src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -22,6 +23,8 @@ namespace Microsoft.AspNetCore.SignalR.StackExchangeRedis;
 /// <typeparam name="THub">The type of <see cref="Hub"/> to manage connections for.</typeparam>
 public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposable where THub : Hub
 {
+    private static readonly CancellationTokenSourcePool _cancellationTokenSourcePool = new();
```

Comment:
This is pretty tricky. I suspect there will be more implementors of `InvokeConnectionAsync` than callers. I'm not sure though. If that's true, should we just move the default timeout to `SingleClientProxy`?